flow/connectors/postgres/postgres.go

```diff
@@ -404,6 +404,8 @@ func pullCore[Items model.Items](
 		if err != nil {
 			return fmt.Errorf("error getting child to parent relid map: %w", err)
 		}
+	} else {
+		childToParentRelIDMap = make(map[uint32]uint32)
```

Comment:
why is empty map okay here? elsewhere we write to this hash, could those writes not happen when this was previously nil?

Reply:
yes, writes to a nil map panic

Reply:
right, but you could do
```go
if map != nil {
  map[k] = v
}
```
my question is if that would be an equivalent fix

Reply:
I prefer checking for nil to avoid code having half filled maps in some states, nil is a very simple edge case in comparison